### PR TITLE
Remove `B` type param: Follow ups

### PR DESCRIPTION
**Pull request by davidpdrsn**
These are some follow-ups to https://github.com/tokio-rs/axum/pull/1778 and https://github.com/tokio-rs/axum/pull/1751. Both those branches should be merged first so this is just a draft.

Basically I think we should use `axum_core::body::Body` everywhere, for consistency and simplicity:

- Add a `Request<B = Body>` type alias.
- Use that type alias everywhere.
- Change `Next<B>` to just `Next`.
- Use `Body` in `axum_core::response::Response` and thus also in `IntoResponse`.
- Remove `BoxBody` and its `boxed` constructor.
- Implement `FromRequest` for `Body`.
- Remove `RawBody` extractor.
- Remove re-exports of `Empty` and `Full`

## TODO

- [x] Should we remove the re-exports of `Full` and `Empty`? `Body` should be able to do all the same things.
- [x] Double check changelog

### CODE REVIEW

File: axum-extra/src/routing/spa.rs
Comment (jplatte) [inline axum-extra/src/routing/spa.rs:1]: Looks like you accidentally restored this module ;)

File: axum-macros/CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 # Unreleased
 
 - **breaking:** `#[debug_handler]` no longer accepts a `body = _` argument. The
+  body type is always `axum::body::Body`
   body type is always `axum::body::Body` ([#1751])
```
Comment (jplatte) [inline axum-macros/CHANGELOG.md:12]: Merge failure.

File: axum/CHANGELOG.md [hunk 2]
```diff
@@ -8,18 +8,19 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 # Unreleased
 
 - **breaking:** The following types/traits are no longer generic over the request body
-  (i.e. the `B` type param has been removed) ([#1751]):
-  - `FromRequest`
+  (i.e. the `B` type param has been removed) ([#1751] and [#1789]):
   - `FromRequestParts`
-  - `Handler`
+  - `FromRequest`
```
Comment (jplatte) [inline axum/CHANGELOG.md:13]: Why reorder them like this?

File: examples/testing/src/main.rs [hunk 3]
```diff
@@ -174,31 +174,7 @@ mod tests {
...
-        assert_eq!(response.status(), StatusCode::OK);
-    }
-
-    // Here we're calling `/requires-connect-into` which requires `ConnectInfo`
-    //
-    // That is normally set with `Router::into_make_service_with_connect_info` but we can't easily
-    // use that during tests. The solution is instead to set the `MockConnectInfo` layer during
-    // tests.
-    #[tokio::test]
-    async fn with_into_make_service_with_connect_info() {
```
Comment (jplatte) [inline examples/testing/src/main.rs:192]: Why remove this? Merge fail?

Comment (davidpdrsn) [reply examples/testing/src/main.rs:192]: Yeah that was just a mistake.

File: ECOSYSTEM.md [hunk 4]
```diff
@@ -32,6 +32,7 @@ If your project isn't listed here and you would like it to be, please feel free
 - [axum-otel-metrics](https://github.com/ttys3/axum-otel-metrics/): A axum OpenTelemetry Metrics middleware with prometheus exporter supported.
 - [jwt-authorizer](https://crates.io/crates/jwt-authorizer): JWT authorization layer for axum (oidc discovery, validation options, claims extraction, etc.) 
 - [axum-typed-multipart](https://crates.io/crates/axum_typed_multipart): Type safe wrapper for `axum::extract::Multipart`.
+- [tower-governor](https://crates.io/crates/tower_governor): A Tower service and layer that provides a rate-limiting backend by [governor](https://crates.io/crates/governor)
```
Comment (davidpdrsn) [inline ECOSYSTEM.md]: I'm probably blind but this also exists on `v0.7.0` (https://github.com/tokio-rs/axum/blob/v0.7.0/ECOSYSTEM.md) so not sure why its shown as "added" here. Same with the two lines above.

File: axum/CHANGELOG.md (hunk 2 above)
Comment (jplatte) [reply axum/CHANGELOG.md:13]: I think `FromRequest` should come before `FromRequestParts` x)

**Review by jplatte — APPROVED** [review]

**Comment by jplatte** [comment]
:shipit:

**Pull request merged**